src/wallet/spend.cpp

```diff
@@ -305,7 +305,7 @@ CoinsResult AvailableCoins(const CWallet& wallet,
 
             std::unique_ptr<SigningProvider> provider = wallet.GetSolvingProvider(output.scriptPubKey);
 
-            int input_bytes = CalculateMaximumSignedInputSize(output, COutPoint(), provider.get(), coinControl);
+            int input_bytes = CalculateMaximumSignedInputSize(output, COutPoint(), provider.get(), wallet.CanGrindR(), coinControl);
```

Comment:
tiny nit:
could cache this result into a variable outside of the loop (after the `only_safe` variable)

Reply:
Good point, I introduced a `const bool` in both places where it's used in a loop.

Reply:
Though perhaps it's overkill, because `IsWalletFlagSet` is just `m_wallet_flags & flag`.

Reply:
yeah, I think that the only place that worth is inside `AvailableCoins`. And not only because of the overhead that introduces (it's one "AND" operation on an atomic variable per wallet UTXO), I was more thinking about the code structure and what we should do if we want to decrease the `cs_wallet` lock contention there.

still, this is still a tiny nit from a person that is thinking about future stuff.